## **Superstore Data Exploration**

The dataset was obtained from Kaggle and contains data about a superstore, including orders, customers, products, sales, discounts, and profit.

In [1]:
import pandas as pd

# Load dataset
data = pd.read_csv("data/superstore_dataset.csv", encoding = "latin1")

### **I. Data Inspection**

Ths structure of the dataset was examined to understand its dimensions, variables, and the type of information in each record.


In [2]:
data.shape

(9994, 21)

In [3]:
data.columns

Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')

The dataset contains 9,994 records and 21 variables covering orders, customers, products, geography, and financial performance. 

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [5]:
data.isnull().sum()

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64

In [6]:
number_duplicates = data.duplicated().sum()
print(number_duplicates)

0


The dataset contains 15 variables stored as the `object` data type and 6 variables stored as `numeric` data types. There are no missing values in any of the 21 variables. Also, there are no duplicate records.

In [7]:
data[["Sales", "Quantity", "Discount", "Profit"]].describe()

,Sales,Quantity,Discount,Profit
count,9994.000000,9994.000000,9994.000000,9994.000000
mean,229.858001,3.789574,0.156203,28.656896
std,623.245101,2.225110,0.206452,234.260108
min,0.444000,1.000000,0.000000,-6599.978000
25%,17.280000,2.000000,0.000000,1.728750
50%,54.490000,3.000000,0.200000,8.666500
75%,209.940000,5.000000,0.200000,29.364000
max,22638.480000,14.000000,0.800000,8399.976000


`Sales`, `Quantity`, `Discount`, `Profit` are numerical variables and contain a range of values that are appropiate for further analysis. `Profit` includes both positive and negative values, showing that some orders generated losses. No data-quality issues were identified from the summary statistics. 

### **II. Data Preparation**

The dataset was prepared to ensure that variables are stored using appropriate data types required for the analysis.

In [8]:
data["Order Date"] = pd.to_datetime(data["Order Date"], format = "%m/%d/%Y")
data["Ship Date"] = pd.to_datetime(data["Ship Date"], format = "%m/%d/%Y")

The `Order Date` and `Ship Date` variables were converted from the object data type to datetime to support analysis.

In [9]:
data["Postal Code"] = data["Postal Code"].astype(str)

The `Postal Code` variable was converted from a numeric data type to object so it is treated as an identifier rather than a numerical measurement.

In [10]:
# Extract the year into a new column
data["Year"] = data["Order Date"].dt.year

The `Year` variable was created to support time-based analysis of business performance. 

### **III. Overall Performance Analysis**

The overall sales and profitability of the superstore were examined to establish a baseline for business performance.


In [11]:
total_sales = data["Sales"].sum()
print(total_sales)

2297200.8603


The superstore had `total sales of $2,297,200.86` across all orders in the dataset. 

In [12]:
total_profit = data["Profit"].sum()
print(total_profit)

286397.0217


The superstore generated a `total profit of $286,397.02` across all orders in the dataset.

In [13]:
profit_margin = (total_profit / total_sales)*100
print(profit_margin)

12.467217240315605


The superstore had an overall profit margin of approximately `12.47%`. Therefore, about $12.47 of profit was generated for every $100 in sales. 

In [14]:
number_orders = data["Order ID"].nunique()
print("Number of unique orders: ", number_orders)
number_customers = data["Customer ID"].nunique()
print("Number of unique customers: ", number_customers)
number_products = data["Product ID"].nunique()
print("Number of unique products sold: ", number_products)

Number of unique orders:  5009
Number of unique customers:  793
Number of unique products sold:  1862


The dataset represents `5,009` unique orders from `793` unique customers, covering `1,862` unique products.

### **IV. Export Prepared Dataset**
The prepared dataset was exported as a separate CSV file for further analysis.

In [15]:
data.to_csv("data/superstore_cleaned_dataset.csv", index = False)